pkg/internalserver/service.go

```diff
@@ -0,0 +1,112 @@
+package internalserver
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"net/http"
+	"time"
+
+	gokitlog "github.com/go-kit/log"
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/influx2cortex/pkg/util/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	DefaultListenAddress           = "0.0.0.0"
+	DefaultListenPort              = 8081
+	DefaultGracefulShutdownTimeout = 5 * time.Second
+)
+
+type ServiceConfig struct {
+	HTTPListenAddress       string
+	HTTPListenPort          int
+	GracefulShutdownTimeout time.Duration
+}
+
+func (c *ServiceConfig) RegisterFlags(flags *flag.FlagSet) {
+	flags.StringVar(&c.HTTPListenAddress, "internalserver.http-listen-address", DefaultListenAddress, "Sets the listen address for the internal http server")
+	flags.IntVar(&c.HTTPListenPort, "internalserver.http-listen-port", DefaultListenPort, "Sets listen address port for the internal http server")
+	flags.DurationVar(&c.GracefulShutdownTimeout, "internalserver.graceful-shutdown-timeout", DefaultGracefulShutdownTimeout, "Graceful shutdown period; defaults to 5 seconds")
```

Comment:
does --help already describe the default? if so then we don't need to repeat it here

Reply:
Oh yea probably - removing that default blurb 👍